src/liboslcomp/CMakeLists.txt

```diff
@@ -16,6 +16,11 @@ endif ()
 
 FLEX_BISON (osllex.l oslgram.y osl lib_src compiler_headers)
 
+if (CMAKE_COMPILER_IS_CLANG OR CMAKE_COMPILER_IS_APPLECLANG OR CMAKE_COMPILER_IS_INTELCLANG)
+    # avoid flex_bison produced warning in osogram.cpp "variable 'osonerrs' set but not used"
+    add_compile_options ("-Wno-unused-but-set-variable")
```

Comment:
I think you should disable warnings like this inside the individual source file if possible, as locally as possible, rather than disabling it for the entire library (unless you really see it popping up endemically). Also, on sufficiently old clang, the warning may be unrecognized?

Reply:
It's bison generated file, so any suggestions for changes to oslgram.y that would achieve that?

Reply:
Example: https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/blob/main/src/liboslcomp/oslgram.y#L31

Reply:
hmmm, that's the same warning, why isn't it working?

Reply:
maybe the `#if` needs to also include icx, if that's where you saw it?